third_party/blink/renderer/modules/ml/webnn/mojo_graph.cc

```diff
@@ -279,61 +279,70 @@ void MojoGraph::ComputeAsyncImpl(const MLNamedArrayBufferViews& inputs,
     exception_state.ThrowDOMException(DOMExceptionCode::kDataError,
                                       "The number of inputs is invalid.");
     return;
   }
   auto named_inputs = ml::webnn::mojom::blink::NamedResources::New(),
        named_outputs = ml::webnn::mojom::blink::NamedResources::New();
-  for (const auto& input : inputs) {
-    String error_message;
-    auto* input_array_buffer_view = input.second.Get();
-    if (input_array_buffer_view == nullptr) {
-      exception_state.ThrowDOMException(DOMExceptionCode::kDataError,
-                                        error_message);
+  {
+    TRACE_EVENT0("blink", "MojoGraph::ComputeSyncImpl::CopyInputs");
+    for (const auto& input : inputs) {
+      String error_message;
+      auto* input_array_buffer_view = input.second.Get();
+      if (input_array_buffer_view == nullptr) {
+        exception_state.ThrowDOMException(DOMExceptionCode::kDataError,
+                                          error_message);
+      }
+      const String& input_name = input.first;
+      auto memory_info = ml::webnn::mojom::blink::MemoryInfo::New();
+      memory_info->byte_offset = inputs_byte_offset_.at(input_name);
+      memory_info->byte_length =
+          input_resources_info_.at(input_name).byte_length;
+      uint8_t* address = inputs_shm_region_.mapping.GetMemoryAs<uint8_t>() +
+                         memory_info->byte_offset;
+      memcpy(address, input_array_buffer_view->BaseAddressMaybeShared(),
+             input_array_buffer_view->byteLength());
+      named_inputs->resources.insert(input_name, std::move(memory_info));
     }
-    const String& input_name = input.first;
-    auto memory_info = ml::webnn::mojom::blink::MemoryInfo::New();
-    memory_info->byte_offset = inputs_byte_offset_.at(input_name);
-    memory_info->byte_length = input_resources_info_.at(input_name).byte_length;
-    uint8_t* address = inputs_shm_region_.mapping.GetMemoryAs<uint8_t>() +
-                       memory_info->byte_offset;
-    memcpy(address, input_array_buffer_view->BaseAddressMaybeShared(),
-           input_array_buffer_view->byteLength());
-    named_inputs->resources.insert(input_name, std::move(memory_info));
+    named_inputs->shared_memory = inputs_shm_region_.region.Duplicate();
   }
-  named_inputs->shared_memory = inputs_shm_region_.region.Duplicate();
   ComputeResult result;
   if (!remote_graph_->Compute(std::move(named_inputs), &result,
                               &named_outputs)) {
     exception_state.ThrowDOMException(DOMExceptionCode::kUnknownError,
                                       "Failed to compute the graph.");
     return;
   };
-  for (const auto& output : outputs) {
-    String error_message;
-    void* output_buffer_address = output.second->BaseAddressMaybeShared();
-    if (output_buffer_address == nullptr) {
-      exception_state.ThrowDOMException(DOMExceptionCode::kOperationError,
-                                        error_message);
-      return;
+  {
```

Comment:
The only suggestion I'd make is to leave some blank space between the sections to help visually demark them.

Reply:
Good point. I'll follow that in the following PRs. Thanks!